src/BenchmarkDotNet/Analysers/RegressionAnalyser.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Mathematics.StatisticalTesting;
+using BenchmarkDotNet.Reports;
+
+namespace BenchmarkDotNet.Analysers
+{
+    public class RegressionAnalyser : AnalyserBase
+    {
+        public static readonly IAnalyser Default = new RegressionAnalyser(RelativeThreshold.Default);
+        
+        public RegressionAnalyser(Threshold threshold) => Threshold = threshold;
+
+        public override string Id => "Regression";
+
+        private Threshold Threshold { get; }
+
+        protected override IEnumerable<Conclusion> AnalyseSummary(Summary summary)
+        {
+            foreach (var sameArguments in summary.BenchmarksCases.GroupBy(benchmark => (benchmark.Descriptor, benchmark.Parameters)))
+            {
+                var benchmarks = sameArguments.ToArray();
+                if (benchmarks.Length != 2)
+                    continue;
+
+                var baseline = benchmarks.Single(benchmark => benchmark.Job.Meta.Baseline);
+                var diff = benchmarks.Single(benchmark => !benchmark.Job.Meta.Baseline);
+                
+                var x = summary[baseline].ResultStatistics.GetOriginalValues().ToArray();
+                var y = summary[diff].ResultStatistics.GetOriginalValues().ToArray();
+
+                var mannWhitneyTestResult = StatisticalTestHelper.CalculateTost(MannWhitneyTest.Instance, x, y, Threshold);
+                var welchTestResult = StatisticalTestHelper.CalculateTost(WelchTest.Instance, x, y, Threshold);
```

Comment:
I don't think that it's a good idea to use Welch's t-test in general case because it's work fine only for distributions which are close to normal. I suggest using only Mann-Whitney.